draft-ietf-httpbis-proxy-status.md

```diff
@@ -352,56 +352,19 @@ This section lists the Proxy Error Types defined by this document. See {{registe
 * Extra Parameters: None.
 * Recommended HTTP status code: 504
 
-### TLS Handshake Error
-
-* Name: tls_handshake_error
-* Description: The intermediary encountered an error during TLS handshake with the next hop.
-* Extra Parameters:
-  - alert_message: a sf-token containing the applicable description string from the TLS Alerts registry.
-* Recommended HTTP status code: 502
-
-### TLS Untrusted Peer Certificate
-
-* Name: tls_untrusted_peer_certificate
-* Description: The intermediary received an untrusted peer certificate during TLS handshake with the next hop.
-* Extra Parameters: None.
-* Recommended HTTP status code: 502
-
-### TLS Expired Peer Certificate
+### TLS Error
 
-* Name: tls_expired_peer_certificate
-* Description: The intermediary received an expired peer certificate during TLS handshake with the next hop.
+* Name: tls_error
+* Description: The intermediary encountered a TLS error when communicating with the next hop.
 * Extra Parameters: None.
 * Recommended HTTP status code: 502
 
-### TLS Unexpected Peer Certificate
-
-* Name: tls_unexpected_peer_certificate
-* Description: The intermediary received an unexpected peer certificate (e.g., SPKI doesn't match) during the TLS handshake with the next hop.
-* Extra Parameters:
-  - identity: a sf-string containing a comma-separated list of Subject Alternative Names from the certificate received from the next hop.
-  - sha256: a sf-string containing the hex-encoded SHA-256 of the certificate received from the next hop.
-  - spki: a sf-string containing the base64-encoded SHA-256 of the Subject Public Key Info (SPKI) from the certificate received from the next hop.
-* Recommended HTTP status code: 502
-
-### TLS Missing Proxy Certificate
+Details of the error should be conveyed in the details parameter.
```

Comment:
So in looking in context, this error stands out as being the only one that explicitly mentions `details`, which might cause some confusion. For example, `http_protocol_error` doesn't mention `details`, yet you use so as part of the illustrative example.

For what it's worth, you *could* split this into two: `tls_protocol_error` and `tls_certificate_error`. The former indicates something went wrong with the TLS protocol (which could include both handshake and ongoing use), while the latter indicates an error while verifying the certificate (for which `details` is useful). Similar to `http_protocol_error`, it serves as both the fallback (e.g. if the proxy doesn't expose certificate errors distinct from TLS errors and treats them just as handshake errors) and as a distinct state (errors sending that may not result in a peer alert - e.g. if it receives a malformed message *from* the peer)

Reply:
Thanks, that's helpful. PTAL.